src/java.base/share/classes/java/lang/AbstractStringBuilder.java

```diff
@@ -1821,4 +1821,115 @@ private final void appendChars(CharSequence s, int off, int end) {
         }
         count += end - off;
     }
+
+    /**
+     * Appends {@code count} copies of the character {@code c} to this sequence.
+     * <p>
+     * The length of this sequence increases by {@code count}.
+     *
+     * @param c      character to append
+     * @param count  number of times to copy
+     *
+     * @return  a reference to this object.
+     *
+     * @since 21
+     * @throws IllegalArgumentException  if {@code count} is less than zero
+     * @throws StringIndexOutOfBoundsException  if the result overflows the buffer
+     */
+    public AbstractStringBuilder repeat(char c, int count) {
+        if (count < 0) {
+            throw new IllegalArgumentException("count is less than zero: " + count);
+        }
+        if (count == 0) {
+            return this;
+        }
+        ensureCapacityInternal(this.count + count);
+        if (isLatin1() && StringLatin1.canEncode(c)) {
+            int index = this.count;
+            while (count-- != 0) {
+                value[index++] = (byte) c;
+            }
```

Comment:
One could reduce to just one running var rather than two:
```
int index = this.count;
int limit = index + count;
for (; index != limit; ++index) {
    value[index] = (byte) c;
}
```

Reply:
True.  Will test to see if there is any performance benefit.

Reply:
Apart from perf, the less "moving parts" there are, the better I personally understand code ;-)